codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -170,6 +170,7 @@ public ChannelFuture writeHeaders(final ChannelHandlerContext ctx, final int str
                                 "Stream %d in unexpected state: %s", stream.id(), stream.state()));
                 }
             }
+            stream.headersSent();
```

Comment:
it may be safer to call this after the write is actually issued to ensure the headers were actually sent (or at least attempted to be sent).